DynamoDS/Dynamo
# MAGN-6241 Regularly publish searchable, easy to read documentation

### Purpose
Link to YouTrack:
[MAGN-6241](http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-6241) 

The purpose of this PR is to create documentation for Dynamo. The process has to be automated and for every push to master, the documentation has to be updated automatically.  

XmlDocToMarkDown was modified to create markdown files which will be deployed to github by mkdocs.

To accomplish this :

1. A new GitHub repo - DynamoAPI was created on DynamoDS.
2. Personal Access Token was generated for DynamoDS 
3. AppVeyor.yml file was updated to run mkdocs for every commit and publish only for merge
4. BuildScript.ps1 was added to BuildScripts folder , which creates the markdown files for the DLL's passed as parameters
5. PublishScript.ps1 was added to BuildScripts folder, which publishes the document to github.

Here is the documentation url :  http://dynamods.github.io/DynamoAPI/

### Workflow
1. For every commit, Appveyor builds Dynamo. 
2. If the build is successful,  then Appveyor runs the script  BuildScript.ps1. This script runs thexmldoc tool and creates the markdown files.
3. If the commit is on master, then Appveyor runs the script PublishScript.ps1. This script installs MkDocs for that instance and then publish the markdown files to github using mkdocs. 

### Running on Desktop

To test this on Desktop,

1. Install MkDocs from http://mkdocs.org/  For windows add the mkdocs to (environment) PATH variable.
2. There is a test library in XmlDocs solution. In XmlDocsToMarkdown project, change the build properties to point DocumentationTestLibrary. It has to be <path to DocumentationTestLibrary.dll> <path to DocumentationTestLibrary.xml>
3. Create methods, events, properties on the test library. Otherwise, it is difficult to debug and identify the issue. 
4. Run XmlDocToMarkDown. Inside debug (or release), you can see the docs folder created. 
5. Now go to the XmlDocToMarkDown project folder, copy the index.md file and paste it inside docs folder (which is in debug (or release).
6 . Open command prompt and run the command Mkdocs serve. 
7.  Go to localhost:8000 and you can see the documentation.

### Declarations


### Reviewers
@ikeough
